From 29dd9ff9b8d3a113bb453f32eddd90974d29c99a Mon Sep 17 00:00:00 2001
From: Sam Mortimer <sam@mortimer.me.uk>
Date: Wed, 15 Aug 2018 21:58:12 -0700
Subject: [PATCH 09/10] [TEMP] fw/b AssetManager: Load lineage resources

*) AssetManager was refactored here:
https://android.googlesource.com/platform/frameworks/base/+/bebfcc46a249a70af04bc18490a897888a142fb8

*) Lineage resources are no longer accessible as a result.  For example, LineageParts crashes from:
08-15 16:22:04.850  2599  2599 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{org.lineageos.lineageparts/org.lineageos.lineageparts.PartsActivity}: android.content.res.Resources$NotFoundException: Resource ID #0x3f080008
[cut]
08-15 16:22:04.850  2599  2599 E AndroidRuntime:        at org.lineageos.internal.notification.LightsCapabilities.supports(LightsCapabilities.java:59)
   which is:
        final int capabilities = context.getResources().getInteger(
                org.lineageos.platform.internal.R.integer.config_deviceLightCapabilities);

*) Until the overall best fix is determined, when framework-res.apk is loaded, load org.linageos.platform-res.pk too.  Works around the above.

Change-Id: I8e3408a1031c0b8e000d78ab59cbcabc860d8b57
---
 core/java/android/content/res/AssetManager.java | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/java/android/content/res/AssetManager.java b/core/java/android/content/res/AssetManager.java
index 289534273d1..ead74b40bc2 100644
--- a/core/java/android/content/res/AssetManager.java
+++ b/core/java/android/content/res/AssetManager.java
@@ -59,6 +59,7 @@ public final class AssetManager implements AutoCloseable {
     private static final boolean DEBUG_REFS = false;
 
     private static final String FRAMEWORK_APK_PATH = "/system/framework/framework-res.apk";
+    private static final String LINEAGE_APK_PATH = "/system/framework/org.lineageos.platform-res.apk";
 
     private static final Object sSync = new Object();
 
@@ -197,6 +198,7 @@ public final class AssetManager implements AutoCloseable {
         try {
             final ArrayList<ApkAssets> apkAssets = new ArrayList<>();
             apkAssets.add(ApkAssets.loadFromPath(FRAMEWORK_APK_PATH, true /*system*/));
+            apkAssets.add(ApkAssets.loadFromPath(LINEAGE_APK_PATH, true /*system*/));
             loadStaticRuntimeOverlays(apkAssets);
 
             sSystemApkAssetsSet = new ArraySet<>(apkAssets);
-- 
2.17.1

